src/main/java/at/fhhagenberg/esd/sqe/ws20/gui/ECCController.java

```diff
@@ -382,19 +388,30 @@ protected void gotoTargetFloor(ActionEvent event) {
         } else
             log(Messages.getString("invalidSelection"));
     }
-
+    
+    private void translateElevator() {
+    	translateElevator(oldPercentage);
+    } 
+    
     private void translateElevator(Integer percentage) {
         double maxHeight = gElevator.getHeight();
         double elevatorHeight = groupElevator.getBoundsInLocal().getHeight();
         double yRect = (maxHeight - elevatorHeight) * percentage / 100.0;
 
         groupElevator.translateYProperty().set(-yRect);
-    }
+        oldPercentage = percentage;
+    }   
 
     private static class FloorState {
         private final BooleanProperty requestUp = new SimpleBooleanProperty(false);
         private final BooleanProperty requestDown = new SimpleBooleanProperty(false);
         private final BooleanProperty stopRequest = new SimpleBooleanProperty(false);
         private final BooleanProperty isServiced = new SimpleBooleanProperty(false);
     }
+    
+    public void setStageAndSetUpListeners(Stage mainStage) {
+    	mainStage.heightProperty().addListener((obs, oldVal, newVal) -> {
```

Comment:
we should fix the codesmell here:

```c++
heightProperty().addListener((obs, oldVal, newVal) -> translateElevator());
```

Reply:
Which code smell sry I don't get it.

Reply:
Ah i see u saw it on Sonar Cloud. Should be fixed now.